backend/middleware/check-auth.js

```diff
@@ -3,16 +3,19 @@ const jwt = require('jsonwebtoken');
 const HttpError = require('../models/http-error');
 
 module.exports = (req, res, next) => {
-  try {
-    const token = req.headers.authorization.split(' ')[1]; // Authorization: 'Bearer TOKEN'
-    if (!token) {
-      throw new Error('Authentication failed!');
+    if (req.method === 'OPTIONS') {
+        return next();
+    }
```

Comment:
This change is important: web browsers make a first request with the OPTIONS verb (not the POST)